app/src/main/java/ch/epfl/qedit/view/answer/MatrixFragment.java

```diff
@@ -0,0 +1,107 @@
+package ch.epfl.qedit.view.answer;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.text.InputFilter;
+import android.text.InputType;
+import android.text.method.DigitsKeyListener;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.WindowManager;
+import android.widget.EditText;
+import android.widget.TableLayout;
+import android.widget.TableRow;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import ch.epfl.qedit.R;
+import ch.epfl.qedit.model.MatrixFormat;
+import java.util.ArrayList;
+
+public class MatrixFragment extends Fragment {
+    private TableLayout tableLayout;
+    private boolean hasDecimal = true;
+    private boolean hasSign = true;
+
+    private ArrayList<TableRow> tableRow = new ArrayList<>();
+    private ArrayList<ArrayList<EditText>> arrayButtons = new ArrayList<>();;
+
+    private int tableRowsNumber = 1;
+    private int tableColumnsNumber = 1;
+    private int maxCharacters = 3;
+    private String hintString;
+    private MatrixFormat matrixFormat;
+    private Context context;
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        matrixFormat = (MatrixFormat) getArguments().getSerializable("m0");
+        setValues();
+    }
+
+    // Used to get the context
+    @Override
```

Comment:
Why do you exactly need these two methods? You don't seem to need the context.

Reply:
Yeah you are right. I thought I needed them but in the end I don't.